Project/Source/StateMachineManager.cpp

```diff
@@ -77,31 +87,43 @@ bool StateMachineManager::UpdateAnimations(GameObject* gameObject, const GameObj
-		result = StateMachineManager::CalculateAnimation(gameObject, owner, currentTimeStatesPrincipal, animationInterpolationsPrincipal, stateMachineResourceUIDPrincipal, currentStatePrincipal, position, rotation, resetSecondaryStatemachine, StateMachineEnum::PRINCIPAL);
+		result = StateMachineManager::CalculateAnimation(gameObject, owner, stateMachineSelected, componentAnimation, position, rotation, resetSecondaryStatemachine);
 		break;
 	case StateMachineEnum::SECONDARY:
-		ResourceStateMachine* resourceStateMachinePrincipal = App->resources->GetResource<ResourceStateMachine>(stateMachineResourceUIDPrincipal);
-		bool secondaryToAnyPrincipal = false; 
-		if (resourceStateMachinePrincipal){
-			secondaryToAnyPrincipal = StateMachineManager::SecondaryEqualsToAnyPrincipal(*currentStateSecondary,resourceStateMachinePrincipal->states);
+		ResourceStateMachine* resourceStateMachinePrincipal = App->resources->GetResource<ResourceStateMachine>(componentAnimation.stateMachineResourceUIDPrincipal);
+		bool secondaryToAnyPrincipal = false;
+		if (resourceStateMachinePrincipal) {
+			secondaryToAnyPrincipal = StateMachineManager::SecondaryEqualsToAnyPrincipal(componentAnimation.currentStateSecondary, resourceStateMachinePrincipal->states);
 		}
-		result = StateMachineManager::CalculateAnimation(gameObject, owner, currentTimeStatesSecondary, animationInterpolationsSecondary, stateMachineResourceUIDSecondary, currentStateSecondary, position, rotation, resetSecondaryStatemachine, StateMachineEnum::SECONDARY, secondaryToAnyPrincipal );
+		result = StateMachineManager::CalculateAnimation(gameObject, owner, stateMachineSelected, componentAnimation, position, rotation, resetSecondaryStatemachine, secondaryToAnyPrincipal);
 		break;
 	}
 
 	return result;
 }
 
-bool StateMachineManager::SecondaryEqualsToAnyPrincipal(const State& currentStateSecondary, const std::unordered_map<UID, State> &states) {
+bool StateMachineManager::SecondaryEqualsToAnyPrincipal(const State& currentStateSecondary, const std::unordered_map<UID, State>& states) {
 	for (const auto& element : states) {
 		if (currentStateSecondary.id == element.second.id) return true;
 	}
 	return false;
 }
 
-bool StateMachineManager::CalculateAnimation(GameObject* gameObject, const GameObject& owner, std::unordered_map<UID, float>& currentTimeStates, std::list<AnimationInterpolation>& animationInterpolations, const UID& stateMachineResourceUID, State* currentState, float3& position, Quat& rotation, bool& resetSecondaryStatemachine,const StateMachineEnum stateMachineEnum, bool principalEqualSecondary) {
+bool StateMachineManager::CalculateAnimation(GameObject* gameObject, const GameObject& owner, StateMachineEnum stateMachineSelected, ComponentAnimation& componentAnimation, float3& position, Quat& rotation, bool& resetSecondaryStatemachine, bool principalEqualSecondary) {
 	bool result = false;
 
+	UID stateMachineResourceUID = componentAnimation.stateMachineResourceUIDPrincipal;
```

Comment:
Same as I asked above. Since this is repeating, maybe a function would be useful if you have these variables as private?